components/sup/src/manager/mod.rs

```diff
@@ -1404,6 +1416,11 @@ impl Manager {
         if let Some(err) = writer.flush().err() {
             warn!("Couldn't flush census state buffer to disk, {}", err);
         }
+        if let Some(err) =
+            perm::set_permissions(&tmp_file, HTTP_GATEWAY_STATE_FILE_PERMISSIONS).err()
+        {
+            warn!("Couldn't set permissions on census state file, {}", err);
```

Comment:
If it's actually important to enforce these permissions, should we continue despite a failure? I think it may be better to abort in this scenario that expose data we don't intend to.

Ditto for the several instances below.

I'd suggest our own implementation to wrap `File::create` that sets permissions as well.

Reply:
I'm not familiar with the rust API, but the `open` system call allows you to set the mode on newly created files which is nice because it prevents the race.  I'll also note it looks like we are using a non-random name for the initial temp file which looks like it doesn't fail if the file already exists, so an attacker could create the file before us and keep an open file handle to it waiting for someone to write to it. Because we are setting the permission on the parent directory thse attacks are probably unlikely I suppose.